lib/utils/getAPI.dart

```diff
@@ -1,29 +1,107 @@
+    }
+    catch(e) {
+      print("Could not connect!");
+    }
+    throw Exception('Could not connect to server');
+  }
+
+  static Future<http.Response> getUser(String url, String auth) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json', HttpHeaders.authorizationHeader: auth};
+    try {
+      String totalurl = '$API_PREFIX$url';
+      response = await http.get(Uri.parse(totalurl), headers: headers);
+      return response;
+    }
+    catch(e) {
+      print(e.toString());
+    }
+    throw Exception('Could not connect to server');
+  }
+
+  static Future<http.Response> logoutUser(String url, String auth) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json', HttpHeaders.authorizationHeader: auth};
+    try {
+      String totalurl = '$API_PREFIX$url';
+      response = await http.get(Uri.parse(totalurl), headers: headers);
+      return response;
+    }
+    catch(e) {
+      print(e.toString());
+    }
+    throw Exception('Could not connect to server');
+  }
+
+  static Future<http.Response> updateUser(String url, String changes) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json', HttpHeaders.authorizationHeader: LocalData.accessToken};
+    try {
+      String totalurl = '$API_PREFIX$url';
+      response = await http.put(Uri.parse(totalurl), body: utf8.encode(changes), headers: headers);
+      return response;
+    }
+    catch(e) {
+      print(e.toString());
+    }
+    throw Exception('Could not connect to server');
+  }
+
+  static Future<http.Response> refreshToken(String url) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json'};
+    try {
+      String totalurl = '$API_PREFIX$url';
+      response = await http.put(Uri.parse(totalurl), body: utf8.encode('{"refreshToken": "${LocalData.refreshToken}"}'), headers: headers);
+      return response;
+    }
+    catch(e) {
+      print(e.toString());
+    }
+    throw Exception('Could not connect to server');
+  }
+
+  static Future<http.Response> deleteUser(String url, String auth) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json', HttpHeaders.authorizationHeader: LocalData.accessToken};
+    try {
+      String totalurl = '$API_PREFIX$url';
+      response = await http.put(Uri.parse(totalurl), headers: headers);
+      return response;
     }
     catch(e) {
       print(e.toString());
     }
-    return ret;
+    throw Exception('Could not connect to server');
   }
 }
 
-class GlobalData {
-  static late int userId;
+class LocalData {
   static late String lastName;
   static late String firstName;
   static late String email;
-}*/
\ No newline at end of file
+  static late String password;
+  static late String accessToken;
+  static late String refreshToken;
+  static late int lastSeen;
```

Comment:
i had read that singletons are not really the greatest to use, but i did change it so that late isnt used for every variable, just one. it should hopefully cause less computations to be made for checking nullity.